homeassistant/components/brunt/__init__.py

```diff
@@ -36,7 +36,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             f"Brunt could not connect with username: {entry.data[CONF_USERNAME]}."
         ) from exc
 
-    async def async_update_data():
+    async def async_update_data() -> dict[str | None, Thing]:
```

Comment:
The key (`thing.serial`) is actually type `str | None` upstream. Although it might not actually be `None` in production. Don't know about that.